### DISCUSSION TITLE
When you use [bg right] I want to make <header> come to the right.

### DISCUSSION THREAD

**Discussion by seop0728** [marp-team/marp#492] [Q&A] [answered]
# ✅ When you use [bg right] I want to make <header> come to the right.

# ✅ My style setting.
```
<style>
header {
 text-align: right;
 }
 ```
 
#  ✅ If you do not use [bg right]
 ```
 ---

<header>
네트워크 토폴로지
</header>

#### 장점
- 구조가 단순하여 설치와 재구성이 쉽고, 장애가 발생해도 복구 시간이 빠르다.
- 각 노드는 바로 이웃하는 장치에만 연결되어 있고, 장치를 추가하거나 삭제할 때 연결선 두 개만 움직이면 된다.
#### 단점
- 하나의 노드에 문제가 발생하면 전체 네트워크를 사용할 수 없다.
- 노드를 재설정하거나 추가 또는 제거하기 위해서는 절차가 복잡하고, 전체 네트워크를 중단해야 된다.

---
```
![image](https://github.com/marp-team/marp/assets/110884476/c0e0de9e-c71e-4047-adf0-004f4fa96573)

# ✅ If you use [bg right]
  
  ```
  ---

<header>
네트워크 토폴로지
</header>

## 1.4) 네트워크 통신방식

### 유니캐스트(Unicast)

- 네트워크에서 가장 일반적으로 사용되는 방식이다. 서버와 클라이언트 간의 일대일(1:1) 통신방식을 말한다.
- 데이터는 목적지 주소를 가진 단일 수신자에게 보내진다.
- 자신의 MAC 주소와 목적지 MAC 주소가 동일하다면 전송된 데이터를 수신하고, 일치하지 않으면 해당 프레임은 버린다.
- 웹 브라우징, 이메일 전송 등에 사용된다.

![bg right:50% 100%](../images/unicast.png)

---
```

![image](https://github.com/marp-team/marp/assets/110884476/788ad3d7-6bb4-4a53-86e1-25324de2834e)

---

# ✅ The result I want
![image](https://github.com/marp-team/marp/assets/110884476/b5070a27-8022-4c9f-b215-e44bac2529e9)

**Comment by yhatt** [comment, marked as answer]
The header contents inserted by the `header` directive is a part of slide contents, and it cannot interfere the split image area that was generated by `![bg right]()`. This behavior is same as [Deckset](https://docs.deckset.com/English.lproj/Media/01-background-images.html#split-slides).

If you want to align the header to the right of the slide, don't use `![bg right]()` and use a normal inline image `![]()` with styling instead. A following Markdown is using `gaia` theme and aligning the image through `float: right`.

```markdown
---
theme: gaia
header: Header contents
---

<style>
header {
 text-align: right;
}
</style>

![](https://marp.app/favicon.png)

# Hello

Your contents are here...

<style scoped>
img:first-of-type {
  float: right;
  width: 50%;
  height: auto;
}
</style>
```

It's also good to put the inline image to the right side of [2 columns layout](https://github.com/orgs/marp-team/discussions/192).

**Reply by yhatt** [reply]
Another solution: #49
